File: backend/src/main/java/cm/imf/pipeline/dto/response/OtpVerifyResponse.java
package cm.imf.pipeline.dto.response;

import java.time.OffsetDateTime;

/**
 * Réponse à la vérification OTP.
 * L'OTP est le seul facteur d'authentification pour tous les rôles sauf SUPER_ADMIN
 * (1ère connexion ou Nième — même flux, toujours AUTHENTICATED).
 */
public record OtpVerifyResponse(

        String  status,
        String  accessToken,
        String  refreshToken,
        String  role,
        String  username,
        String  imfUid,
        String  imfCode,
        String  imfNom,
        boolean mustChangePassword,
        Long    expiresIn,
        OffsetDateTime sessionExpiresAt

) {
    public static final String AUTHENTICATED = "AUTHENTICATED";

    public static OtpVerifyResponse authenticated(AuthResponse auth) {
        return authenticated(auth, null);
    }

    public static OtpVerifyResponse authenticated(AuthResponse auth, OffsetDateTime sessionExpiresAt) {
        return new OtpVerifyResponse(
                AUTHENTICATED,
                auth.accessToken(), auth.refreshToken(),
                auth.role(), auth.username(),
                auth.imfUid(), auth.imfCode(), auth.imfNom(),
                auth.mustChangePassword(),
                auth.expiresIn(),
                sessionExpiresAt
        );
    }
}

File: mobile/lib/core/models/otp_verify_response.dart
class OtpVerifyResponse {
  final String status;
  final String accessToken;
  final String refreshToken;
  final String role;
  final String username;
  final String? imfUid;
  final String? imfCode;
  final String? imfNom;
  final int expiresIn;

  OtpVerifyResponse({
    required this.status,
    required this.accessToken,
    required this.refreshToken,
    required this.role,
    required this.username,
    this.imfUid,
    this.imfCode,
    this.imfNom,
    required this.expiresIn,
  });

  factory OtpVerifyResponse.fromJson(Map<String, dynamic> json) {
    return OtpVerifyResponse(
      status: json['status'] as String? ?? 'AUTHENTICATED',
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
      role: json['role'] as String,
      username: json['username'] as String,
      imfUid: json['imfUid'] as String?,
      imfCode: json['imfCode'] as String?,
      imfNom: json['imfNom'] as String?,
      expiresIn: (json['expiresIn'] as num?)?.toInt() ?? 3600,
    );
  }
}
